x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -283,15 +288,34 @@ public void authenticate(final AuthenticationToken authenticationToken, final Ac
             }
 
             // Validate JWT: Extract JWT and claims set, and validate JWT.
-            jwtAuthenticator.authenticate(
-                jwtAuthenticationToken,
-                ActionListener.wrap(claimsSet -> processValidatedJwt(tokenPrincipal, jwtCacheKey, claimsSet, listener), ex -> {
-                    final String msg = "Realm [" + name() + "] JWT validation failed for token=[" + tokenPrincipal + "].";
-                    logger.debug(msg, ex);
-                    // TODO: No point to continue to another realm if failure is ParseException
-                    listener.onResponse(AuthenticationResult.unsuccessful(msg, ex));
-                })
-            );
+            jwtAuthenticator.authenticate(jwtAuthenticationToken, ActionListener.wrap(claimsSet -> {
+                logger.debug(
+                    "Realm [{}] JWT validation success for token=[{}] with header [{}] and claimSet [{}]",
+                    name(),
+                    tokenPrincipal,
+                    jwtAuthenticationToken.getSignedJWT().getHeader(),
+                    jwtAuthenticationToken.getJWTClaimsSet()
+                );
```

Comment:
We should either wrap it inside a `isDebugEnabled` check or use the method with `Supplier` parameter since the claimSet could be large and expensive (on a happy path) to format to a string.